src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClientTest.java

```diff
@@ -20,4 +77,132 @@ public void testParsePullRequestMergeStatus() throws Exception {
     assertThat(resp.getConflicted(), is(false));
     assertThat(resp.getVetoes(), hasSize(1));
   }
+
+  @Test
+  public void getPullRequests_gets_empty_list() throws Exception {
+    JSONObject json = new JSONObject();
+    json.put("isLastPage", true);
+    json.put("values", new JSONArray());
+
+    stubFor(get(pullRequestPath(0)).willReturn(okJson(json.toString())));
+
+    assertThat(client.getPullRequests(), is(empty()));
+  }
+
+  @Test
+  public void getPullRequests_gets_pull_requests_from_multiple_pages() throws Exception {
+    final int totalPullRequests = 27;
+    final int pageSize = 10;
+    int pullRequestCounter = 0;
+    boolean isLastPage = false;
+
+    while (!isLastPage) {
+      JSONObject json = new JSONObject();
+      JSONArray jsonArray = new JSONArray();
+      final int thisPageStart = pullRequestCounter;
+
+      for (int i = 0; i < pageSize; ++i) {
+        jsonArray.add(new JSONObject());
+        if (++pullRequestCounter == totalPullRequests) {
+          isLastPage = true;
+          break;
+        }
+      }
+
+      json.put("values", jsonArray);
+      json.put("isLastPage", isLastPage);
+      if (!isLastPage) {
+        json.put("nextPageStart", pullRequestCounter);
+      }
+
+      stubFor(get(pullRequestPath(thisPageStart)).willReturn(okJson(json.toString())));
+    }
+
+    assertThat(client.getPullRequests(), hasSize(totalPullRequests));
+  }
+
+  @Test
+  public void getPullRequests_throws_on_error_404() throws Exception {
+    stubFor(get(pullRequestPath(0)).willReturn(notFound()));
+
+    expectedException.expect(RuntimeException.class);
+    expectedException.expectMessage(containsString("Didn't get a 200 response from Stash PR GET"));
+
+    client.getPullRequests();
+  }
+
+  @Test
+  public void getPullRequestComments_gets_empty_list() throws Exception {
+    JSONObject json = new JSONObject();
+    json.put("isLastPage", true);
+    json.put("values", new JSONArray());
+
+    stubFor(get(pullRequestActivitiesPath(0)).willReturn(okJson(json.toString())));
+
+    assertThat(
+        client.getPullRequestComments(projectName, repositoryName, pullRequestId), is(empty()));
+  }
+
+  @Test
+  public void getPullRequestComments_gets_pull_requests_from_multiple_pages() throws Exception {
+    final int totalComments = 27;
+    final int pageSize = 10;
+    int commentCounter = 0;
+    boolean isLastPage = false;
+
+    while (!isLastPage) {
```

Comment:
I think this has to be extracted to a helper method, or we should instead write the data in a more direct ways

Reply:
I don't envision more tests using multi-page data. We can make make those two cases share some code, but I'm not sure it would be an improvement. There are subtle differences between `getPullRequests` and `getPullRequestComments`, and the common code would need to take care of those differences. That would only add complexity.

I tried JSON in strings first, but it's hard to read without formatting.

I'm thinking of loading JSON from files.

Reply:
I wasn't thinking about code reuse, just about moving the logic out of main test body to improve readability